hphp/runtime/base/runtime-option.cpp

```diff
@@ -487,6 +492,14 @@ static inline bool alignMacroFusionPairs() {
   return false;
 }
 
+static inline bool armLseDefault() {
+#ifdef __linux__
+  return (getauxval(AT_HWCAP) & HWCAP_ATOMICS) != 0;
```

Comment:
Agreed with @pinski-cavium.  Strange that `armLseDefault()` would not check whether the `arm` part holds.